model/config.go

```diff
@@ -329,113 +346,94 @@ func MakeIstioStore(store ConfigStore) IstioConfigStore {
 	})
-
-	// project to rules
-	out := make([]*proxyconfig.RouteRule, len(rules))
-	for i, rule := range rules {
-		out[i] = rule.Spec
-	}
-	return out
 }
 
-func (i *istioConfigStore) RouteRulesByDestination(instances []*ServiceInstance) []*proxyconfig.RouteRule {
-	type config struct {
-		Key  string
-		Spec *proxyconfig.RouteRule
+func (store *istioConfigStore) RouteRules(instances []*ServiceInstance, destination string) []Config {
+	out := make([]Config, 0)
+	configs, err := store.List(RouteRule.Type, NamespaceAll)
+	if err != nil {
+		return nil
 	}
-	rules := make([]config, 0)
 
-	for key, rule := range i.RouteRules() {
-		for _, instance := range instances {
-			if rule.Destination == instance.Service.Hostname {
-				rules = append(rules, config{Key: key, Spec: rule})
-				break
-			}
+	for _, config := range configs {
+		rule := config.Spec.(*proxyconfig.RouteRule)
+
+		// validate that rule match predicate applies to destination service
+		hostname := ResolveHostname(config.ConfigMeta, rule.Destination)
+		if hostname != destination {
+			continue
 		}
-	}
 
-	// sort by high precedence first, key string second (keys are unique)
-	sort.Slice(rules, func(i, j int) bool {
-		return rules[i].Spec.Precedence > rules[j].Spec.Precedence ||
-			(rules[i].Spec.Precedence == rules[j].Spec.Precedence &&
-				rules[i].Key < rules[j].Key)
-	})
+		// validate that rule match predicate applies to source service instances
+		if rule.Match != nil && !MatchSource(config.ConfigMeta, rule.Match.Source, instances) {
+			continue
+		}
 
-	// project to rules
-	out := make([]*proxyconfig.RouteRule, len(rules))
-	for i, rule := range rules {
-		out[i] = rule.Spec
+		out = append(out, config)
 	}
+
 	return out
 }
 
-func (i *istioConfigStore) IngressRules() map[string]*proxyconfig.IngressRule {
-	out := make(map[string]*proxyconfig.IngressRule)
-	rs, err := i.List(IngressRule.Type, "")
+func (store *istioConfigStore) RouteRulesByDestination(instances []*ServiceInstance) []Config {
+	out := make([]Config, 0)
+	configs, err := store.List(RouteRule.Type, NamespaceAll)
 	if err != nil {
-		glog.V(2).Infof("IngressRules => %v", err)
+		return nil
 	}
-	for _, r := range rs {
-		if rule, ok := r.Spec.(*proxyconfig.IngressRule); ok {
-			out[r.Key()] = rule
+
+	for _, config := range configs {
+		rule := config.Spec.(*proxyconfig.RouteRule)
+		destination := ResolveHostname(config.ConfigMeta, rule.Destination)
+		for _, instance := range instances {
+			if destination == instance.Service.Hostname {
+				out = append(out, config)
+				break
+			}
 		}
 	}
+
 	return out
 }
 
-func (i *istioConfigStore) EgressRules() map[string]*proxyconfig.EgressRule {
+func (store *istioConfigStore) EgressRules() map[string]*proxyconfig.EgressRule {
 	out := make(map[string]*proxyconfig.EgressRule)
-	rs, err := i.List(EgressRule.Type, "")
+	rs, err := store.List(EgressRule.Type, "")
 	if err != nil {
-		glog.V(2).Infof("EgressRules => %v", err)
```

Comment:
Are you logging this info in another place ? It is quite useful to see the values (maybe at v > 2 ).

Reply:
It's not useful to see in an infinite loop where this runs